src/components/ui/Panel.jsx

```diff
@@ -91,41 +96,41 @@ export default class Panel extends React.Component {
    */
   stopResize = event => {
     event.preventDefault();
-    clearTimeout(this.timer);
 
-    const nativeEvent = event.nativeEvent;
-    const clientY = nativeEvent.changedTouches
-      ? nativeEvent.changedTouches[0].clientY
-      : nativeEvent.clientY;
-    const sizes = [0, window.innerHeight / 2, window.innerHeight];
-    const positionIndex = getClosestIndex(sizes, clientY);
+    document.removeEventListener('mousemove', this.moveCallback);
+    document.removeEventListener('touchmove', this.moveCallback);
+
+    const { previousSize, currentHeight } = this.state;
 
     let size = null;
-    if (positionIndex === 2 && this.holding) {
-      size = 'minimized';
-    } else if (positionIndex === 1 && !this.holding) {
-      size = 'minimized';
-    } else if (positionIndex === 0) {
-      size = 'maximized';
+    if (event.timeStamp - this.interactionStarted < CLICK_RESIZE_TIMEOUT_MS) {
+      // the resize handler was only clicked, provide 'smart' resize
+      size = !previousSize ? 'minimized' : null;
+    } else {
+      // the resize handler was really dragged-n-dropped
+      const sizes = [0, window.innerHeight / 2, window.innerHeight];
+      const positionIndex = getClosestIndex(sizes, currentHeight);
+      if (positionIndex === 0) {
+        size = 'minimized';
+      } else if (positionIndex === 2) {
+        size = 'maximized';
+      }
     }
 
-    this.holding = false;
     this.setState({
-      isTransitioning: true,
+      holding: false,
       size,
       currentHeight: null,
+      isTransitioning: true,
```

Comment:
I thing `isTransitioning` is not used anymore ?

Reply:
It is for now. We need to call `updateMobileMapUI` after a CSS transition is done on the panel. It's currently done through `componentDidUpdate` when this state value is changed (see the lines below).
But you're right to spot it. It's an old hack with ugly coupling between CSS and JS… I'll try find an alternative using https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/transitionend_event